src/lib/storage/table.cpp

```diff
@@ -160,12 +174,11 @@ void Table::append_chunk(const Segments& segments, const std::optional<Polymorph
         break;
     }
   }
-#endif
 
-  std::shared_ptr<MvccData> mvcc_data;
+  DebugAssert(!mvcc_data || mvcc_data->size() == chunk_size, "Invalid MvccData size");
+#endif
 
-  if (_use_mvcc == UseMvcc::Yes) {
-    // append_chunk is a helper method and rows are made visible immediately
+  if (_use_mvcc == UseMvcc::Yes && !mvcc_data) {
     mvcc_data = std::make_shared<MvccData>(chunk_size, CommitID{0});
```

Comment:
This line will only be executed in release as the 2nd `DebugAssert` will fail in debug.
I would either remove the 2nd `DebugAssert` or this line.

Reply:
this line it is, implicit allocations of significant amounts of memory should not happen and were what lead to the perf impact fixed by this PR.